core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataCreateIndexService.java

```diff
@@ -514,6 +516,14 @@ private void validate(CreateIndexClusterStateUpdateRequest request, ClusterState
     }
 
     public void validateIndexSettings(String indexName, Settings settings) throws IndexCreationException {
+        List<String> validationErrors = indexSettingsValidation(settings);
+        if (validationErrors.isEmpty() == false) {
+            throw new IndexCreationException(new Index(indexName),
+                    new IllegalArgumentException(getMessage(validationErrors)));
+        }
+    }
+
+    public List<String> indexSettingsValidation(Settings settings) {
```

Comment:
Agree. I've pushed a [commit](https://github.com/jasontedor/elasticsearch/commit/7135e14514a568177cd2a9a1038f083eeec83959) that reflects these suggestions.